eva-accession-clustering/src/test/java/uk/ac/ebi/eva/accession/clustering/batch/io/clustering_writer/RemappedVariantsClusteringWriterTest.java

```diff
@@ -339,137 +339,173 @@ public void test_variant_not_clustered_rs_id_exist() throws Exception {
-     * 306  ASM2_Chr1_1000_INS	ASM2	1000	Chr1	INS
-     * 400  ASM2_Chr1_1500_INS	ASM2	1500	Chr1	INS
+     *              SubmittedVariantOperationEntity
+     * ACCESSION    EVENT_TYPE          REASON                  INACTIVe_OBJECTS
+     * 306          RS_SPLIT_CANDIDATE  Hash mismatch with 306  {ss-500 and ss-501}
      */
     @Test
     @DirtiesContext
-    public void test_clustered_variant_remapped_RS_split_diff_location() throws Exception {
-        //original submitted variants
-        SubmittedVariantEntity submittedVariantEntity1 = getSubmittedVariantEntity(ASM_1, PROJECT_1, 1000, 3000000006L,
-                                                                                   5000000001L, "", "TT", NOT_REMAPPED);
-        mongoTemplate.insert(submittedVariantEntity1, SUBMITTED_VARIANT_COLLECTION);
-        SubmittedVariantEntity submittedVariantEntity2 = getSubmittedVariantEntity(ASM_1, PROJECT_1, 1000, 3000000006L,
-                                                                                   5000000000L, "A", "AT",
-                                                                                   NOT_REMAPPED);
-        mongoTemplate.insert(submittedVariantEntity2, SUBMITTED_VARIANT_COLLECTION);
-
-        //clustered variants
-        ClusteredVariantEntity clusteredVariantEntity1 = getClusteredVariantEntity(ASM_1, 1000, 3000000006L,
-                                                                                   VariantType.INS);
-        mongoTemplate.insert(clusteredVariantEntity1, CLUSTERED_VARIANT_COLLECTION);
-
-        //submitted variants after remapping
-        SubmittedVariantEntity submittedVariantEntityRemapped1 = getSubmittedVariantEntity(ASM_2, PROJECT_1, 1500, 3000000006L,
-                                                                                   5000000001L, "", "TT", ASM_1);
+    public void test_detect_RS_MERGE_AND_SPLIT_different_batches() throws Exception {
+        //Batch 1
+        SubmittedVariantEntity submittedVariantEntityRemapped1 = getSubmittedVariantEntity(ASM_2, "project_1",
+                5000000001L, 3000000001L, 1000,"A", "T", ASM_1);
         mongoTemplate.insert(submittedVariantEntityRemapped1, SUBMITTED_VARIANT_COLLECTION);
-        SubmittedVariantEntity submittedVariantEntityRemapped2 = getSubmittedVariantEntity(ASM_2, PROJECT_1, 1000, 3000000006L,
-                                                                                   5000000000L, "", "T", ASM_1);
-        mongoTemplate.insert(submittedVariantEntityRemapped2, SUBMITTED_VARIANT_COLLECTION);
+        SubmittedVariantEntity submittedVariantEntityRemapped3 = getSubmittedVariantEntity(ASM_2, "project_3",
+                5000000003L, 3000000003L, 2000,"A", "T", ASM_1);
+        mongoTemplate.insert(submittedVariantEntityRemapped3, SUBMITTED_VARIANT_COLLECTION);
 
-        //asm2 clustered
         List<SubmittedVariantEntity> submittedVariantEntityList = new ArrayList<>();
```

Comment:
Not urgent bust just a note: This list primarily seems to be just a holding area for a collection of objects.  You can use Arrays.asList or Collections.singletonList to quickly create lists of arbitrary object instead of clearing and adding to this list object every time.

Reply:
good point. updated